src/nc_ssl.c

```diff
@@ -277,6 +278,13 @@ nc_ssl_read(SSL *ssl, void *buf, size_t num) {
                 return NC_ERROR;
             }
 
+            if (code == SSL_ERROR_WANT_READ) {
```

Comment:
This comes after the blocking for a read or write. It doesn’t sound like that’s what you want. Should it happen before the if on line 273?

Reply:
It's the looping that is in effect blocking, not the actual blocking above this line. This is indeed in the right place.